File: examples/java/spring-boot/two_consumers_one_provider/provider/src/main/java/br/com/zup/pact/provider/service/impl/AccountServiceImpl.java
package br.com.zup.pact.provider.service.impl;

import br.com.zup.pact.provider.dto.AccountDetailsDTO;
import br.com.zup.pact.provider.dto.BalanceDTO;
import br.com.zup.pact.provider.repository.AccountRepository;
import br.com.zup.pact.provider.service.AccountService;
import java.util.List;
import java.util.Optional;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class AccountServiceImpl implements AccountService {

    private final AccountRepository accountRepository;

    @Override
    public Optional<AccountDetailsDTO> getAccountDetailsByClientId(Integer clientId) {
        return accountRepository.findByClientId(clientId);
    }

    @Override
    public Optional<List<AccountDetailsDTO>> getAll() {
        return accountRepository.getAll();
    }

    @Override
    public Optional<BalanceDTO> getBalanceByClientId(Integer clientId) {
        return accountRepository.getBalanceByClientId(clientId);
    }
}

File: example/kotlin/one-consumer-one-provider/provider/src/main/kotlin/br/com/zup/pact/provider/service/AccountServiceImpl.kt
package br.com.zup.pact.provider.service

import br.com.zup.pact.provider.dto.AccountDetailsDTO
import br.com.zup.pact.provider.dto.BalanceDTO
import br.com.zup.pact.provider.repository.AccountRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class AccountServiceImpl(@Autowired val accountRepository: AccountRepository): AccountService {
    override fun getAccountDetailsByClientId(clientId: Int): AccountDetailsDTO? {
        return accountRepository.findByClientId(clientId)
    }

    override fun getAll(): List<AccountDetailsDTO> {
        return accountRepository.getAllAccounts()
    }

    override fun getBalanceByClientId(clientId: Int): BalanceDTO? {
        return accountRepository.getBalanceByClientId(clientId)
    }
}
